solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml

```diff
@@ -22,6 +22,10 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-autoconfigure</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-actuator</artifactId>
+		</dependency>
```

Comment:
Actuator should be marked an optional dependency.

If users want to use the optional health indicators/contributors, then they should add actuator in their application's POM. This was the pattern I've seen other Spring libraries follow.

The core of this project (i.e. the properties auto-configuration) doesn't require actuator to function.

Reply:
### Short response:
It should not be marked as an optional dependency.

### Long Explanation:
First of all, I would like to advise you to go back and take a look at the PR where we had same discussion about optional dependencies for Connector Framework - https://github.com/SolaceDev/solace-connector-framework/pull/145.

Second thing, let us define where are we heading with this product to. There are 2 different options:
- we define the scope of this product for developers
- we define the scope of this product for end users

Going further with `optional` status for dependency we allow developers, who will be building the product to exclude it completely from the final build. While this is fine to stay flexible and being able to create eventually many different versions of the same product based on the granularity possibilities provided to us with such a dependency management, that's only fine for development side, but not for product. As the huge drawback we get the product fragmentation, where we/someone else need to build many different versions in case of enabling or disabling some features.

The second concept, which is the concept what we want to bring, is the concept of [feature toggling](https://martinfowler.com/articles/feature-toggles.html), introduced by Martin Fowler, allowing us to operate by flags, which enable or disable some functionality without necessity of rebuilding the product or issuing a new version. Moving that concept further is impossible if any of the product won't have necessary functionality included into the build.

`spring-boot-starter-actuator` is a required dependency for Health Indicators and Health Contributors as all Health Indicators and Health Contributors use the nested `spring-boot-actuator` dependency.
```
 +- org.springframework.boot:spring-boot-starter-actuator:jar:3.1.4:compile
### Short response:
It should not be marked as an optional dependency.

### Long Explanation:
First of all, I would like to advise you to go back and take a look at the PR where we had same discussion about optional dependencies for Connector Framework - https://github.com/SolaceDev/solace-connector-framework/pull/145.

Second thing, let us define where are we heading with this product to. There are 2 different options:
- we define the scope of this product for developers
- we define the scope of this product for end users

Going further with `optional` status for dependency we allow developers, who will be building the product to exclude it completely from the final build. While this is fine to stay flexible and being able to create eventually many different versions of the same product based on the granularity possibilities provided to us with such a dependency management, that's only fine for development side, but not for product. As the huge drawback we get the product fragmentation, where we/someone else need to build many different versions in case of enabling or disabling some features.

The second concept, which is the concept what we want to bring, is the concept of [feature toggling](https://martinfowler.com/articles/feature-toggles.html), introduced by Martin Fowler, allowing us to operate by flags, which enable or disable some functionality without necessity of rebuilding the product or issuing a new version. Moving that concept further is impossible if any of the product won't have necessary functionality included into the build.

`spring-boot-starter-actuator` is a required dependency for Health Indicators and Health Contributors as all Health Indicators and Health Contributors use the nested `spring-boot-actuator` dependency.
```
[INFO] +- org.springframework.boot:spring-boot-starter-actuator:jar:3.1.4:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter:jar:3.1.4:compile
[INFO] |  |  \- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[INFO] |  +- org.springframework.boot:spring-boot-actuator-autoconfigure:jar:3.1.4:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-actuator:jar:3.1.4:compile
```

All those Indicators together with Contributors are now a part of the the codebase, which will be used in **all cases** in Solace Spring Cloud Stream Project, thus it would be illogically to make such a dependency optional and then have to re-include same dependency on the higher level every time we need it. 

On your comment about Spring, please pay attention, that it is not possible to apply the same approach to everything, thus even Spring ecosystem might use it, Spring itself doesn't bring end product for customers, it brings a product for developers, requiring a high level of granularity to build end product they need.

As for us - we build such products with a note in mind that enabling or disabling some pieces of functionality should not require a full rebuild, but just enabling or disabling a configuration flag in configuration file.[INFO] |  +- org.springframework.boot:spring-boot-starter:jar:3.1.4:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-logging:jar:3.1.4:compile
[INFO] |  |  |  +- ch.qos.logback:logback-classic:jar:1.4.11:compile
[INFO] |  |  |  |  \- ch.qos.logback:logback-core:jar:1.4.11:compile
[INFO] |  |  |  +- org.apache.logging.log4j:log4j-to-slf4j:jar:2.19.0:compile
[INFO] |  |  |  |  \- org.apache.logging.log4j:log4j-api:jar:2.19.0:compile
[INFO] |  |  |  \- org.slf4j:jul-to-slf4j:jar:2.0.9:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[INFO] |  |  \- org.yaml:snakeyaml:jar:2.2:compile
[INFO] |  +- org.springframework.boot:spring-boot-actuator-autoconfigure:jar:3.1.4:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-actuator:jar:3.1.4:compile
```

All those Indicators together with Contributors are now a part of the the codebase, which will be used in **all cases** in Solace Spring Cloud Stream Project, thus it would be illogically to make such a dependency optional and then have to re-include same dependency on the higher level every time we need it. 

On your comment about Spring, please pay attention, that it is not possible to apply the same approach to everything, thus even Spring ecosystem might use it, Spring itself doesn't bring end product for customers, it brings a product for developers, requiring a high level of granularity to build end product they need.

As for us - we build such products with a note in mind that enabling or disabling some pieces of functionality should not require a full rebuild, but just enabling or disabling a configuration flag in configuration file.

Reply:
> First of all, I would like to advise you to go back and take a look at the PR where we had same discussion about optional dependencies for Connector Framework - https://github.com/SolaceDev/solace-connector-framework/pull/145.

I don't think think this is relevant.

This project's use case isn't just for connector framework, nor is it for Solace Spring Cloud. It's to allow developers to add auto-wiring capabilities to their applications. We can't force them to add endpoints to their application that they don't need.

> On your comment about Spring, please pay attention, that it is not possible to apply the same approach to everything, thus even Spring ecosystem might use it, Spring itself doesn’t bring end product for customers, it brings a product for developers, requiring a high level of granularity to build end product they need.

> As for us - we build such products with a note in mind that enabling or disabling some pieces of functionality should not require a full rebuild, but just enabling or disabling a configuration flag in configuration file.

This is incorrect as per my previous comment. This is "a product for developers, requiring a high level of granularity to build end product they need". That's exactly what this project is.